Source/Core/DiscIO/DiscUtils.h

```diff
@@ -61,9 +61,9 @@ constexpr int PREFERRED_MIN_BLOCK_SIZE = 0x8000;
 // For performance reasons, blocks shouldn't be too large.
 constexpr int PREFERRED_MAX_BLOCK_SIZE = 0x200000;
 
+// If we didn't find a good GCZ block size, pick the block size which was hardcoded
+// in legacy versions. That way, at least we ensure compatibility.
```

Comment:
This does not ensure compatibility. (Yes, old versions of Dolphin could create files which couldn't be correctly read by itself.) The "not worse than older versions" wording was chosen for a reason.